docs/layout-of-source-files.rst

```diff
@@ -216,101 +216,562 @@ the code below creates new global symbols ``alias`` and ``symbol2`` which refere
+`The Remix IDE <https://remix.ethereum.org/>`_ uses this mechanism to allow files to be imported
+directly from github:
+
+.. code-block:: solidity
+    :caption: contract.sol
+
+    import "https://github.com/ethereum/dapp-bin/library/iterable_mapping.sol" as it_mapping;
+
+When compiling locally you can use import remapping to replace the protocol and domain part with a
+local path:
 
 .. code-block:: bash
 
-  solc module1:github.com/ethereum/dapp-bin/=/usr/local/dapp-bin/ \
-       module2:github.com/ethereum/dapp-bin/=/usr/local/dapp-bin_old/ \
-       source.sol
+    solc :https://github.com/ethereum/dapp-bin=/usr/local/dapp-bin contract.sol
 
-This means that all imports in ``module2`` point to the old version but imports
-in ``module1`` point to the new version.
+Note the leading ``:``.
+It is necessary when the remapping context is empty.
+Otherwise the ``https:`` part would be interpreted by the compiler as the context.
 
 .. note::
 
-  ``solc`` only allows you to include files from certain directories. They have
-  to be in the directory (or subdirectory) of one of the explicitly specified
-  source files or in the directory (or subdirectory) of a remapping target. If
-  you want to allow direct absolute includes, add the remapping ``/=/``.
+    When remapping, keep in mind that the prefix must match exactly.
+    ``https://example.com/project=/project`` will match  ``https://example.com/project/contract.sol``
+    but not ``example.com/project/contract.sol``, ``https://example.com/project///contract.sol`` or
+    ``https://EXAMPLE.COM/project/contract.sol``.
 
-If there are multiple remappings that lead to a valid file, the remapping
-with the longest common prefix is chosen.
+    Also, since URLs look to the compiler just like imports relative to base there is no
+    normalization involved.
+    The import key for ``EXAMPLE.COM/project///contract.sol`` is exactly
+    ``EXAMPLE.COM/project///contract.sol`` and not ``https://example.com/project/contract.sol``.
+    It will only get normalized if the compiler passes the key to the file loader but then the
+    normalization rules for paths, not URLs will be applied.
 
-**Remix**:
+.. note::
 
-`Remix <https://remix.ethereum.org/>`_ provides an automatic remapping for
-GitHub and automatically retrieves the file over the network. You can import
-the iterable mapping as above,  e.g.
+    ``file://`` prefix is stripped from import paths and from filesystem paths specified in
+    ``urls`` in Standard JSON. It is **not** stripped from filesystem paths provided on the command
+    line.
+    For example the following will not result in ``contract.sol`` being loaded:
 
-::
+        .. code-block:: bash
+
+            solc file://contract.sol
+
+    The compiler will instead try to find it in a directory called ``file:`` and fail if such a
+    directory does not exist or does not contain ``contract.sol``.
```

Comment:
I think we should strip `file://` from command-line paths too if we're stripping it in other situations.